### DISCUSSION TITLE
Raspberry pi baremetal port of 86Box

### DISCUSSION THREAD

**Discussion by ArnoldUK** [86Box/86Box#3240] [Ideas]
**Is your feature request related to a problem? Please describe.**
See title

**Describe the solution you'd like**
Would be great to see a Raspberry pi ARM release in the future that could run directly on the hardware without the linux OS subsystem. I know it's possible as there are many emulators that run baremetal using the [circle ](https://github.com/rsta2/circle) library.

Would it be as simple as stripping out all the SDL and QT stuff and routing it through the [circle ](https://github.com/rsta2/circle) library ? I know it would not have much of a blingy gui and restricted to maybe only one machine type with limited hardware. Only a thought though and not expecting anything to happen soon.

**Describe alternatives you've considered**
I'm using a fork of [Fake86 ](https://github.com/lgblgblgb/fake86) now cross-compiled and re-named [Faux86 ](https://github.com/jhhoward/Faux86) which works pretty well but the project has not been updated for a few years.

**Comment by chungy** [comment]
Linux isn't a "subsystem" on the Pi, it's literally the operating system. The 86Box binary, compiled natively, already runs pretty much as bare-metal as you want it to. If performance is the key issue you are after, might consider a minimalist window manager (eg: i3) and maybe a realtime-enabled kernel that lets you put 86Box in realtime priority (this has some other caveats, it's likely overboard).

86Box is fairly dependent on having a file system and some other nice features that an operating system provides; features that would otherwise need to be reimplemented from scratch.

**Reply by ArnoldUK** [reply]
> Linux isn't a "subsystem" on the Pi, it's literally the operating system. The 86Box binary, compiled natively, already runs pretty much as bare-metal as you want it to. If performance is the key issue you are after, might consider a minimalist window manager (eg: i3) and maybe a realtime-enabled kernel that lets you put 86Box in realtime priority (this has some other caveats, it's likely overboard).
> 
> 86Box is fairly dependent on having a file system and some other nice features that an operating system provides; features that would otherwise need to be reimplemented from scratch.

Sorry, you misunderstood me, the Linux kernel provides a sub-system for 86Box to run on as it will not run independently without it. Yes, an OS like Linux or Windows is required to run 86Box but it does not run baremetal which is what I was suggesting. A baremetal port of 86Box would not depend on sub parts of an OS to run and it would have it's own dedicated kernel for communicating with the RPi hardware.
This is possible through the rsta2 circle library like I suggested but would require an interface to be built, directing all IO calls to the circle kernel SDK. 

Performance is not an issue and yes a very minimal custom linux kernel could speed up boots times, but nothing close to the performance achieved running baremetal on say a RPi4 without all the bloat of an OS running in the background.

**Comment by symbios24** [comment]
Hi,it will be great to make it work in a raspberry pi 4,i been trying with no success to build it or run the appimage with no luck
